R/execution.r

```diff
@@ -92,12 +92,20 @@ execute = function(request) {
     handle_message = identity
     handle_warning = identity
   } else {
-    handle_value = function (obj) {
-        data = list()
-        data['text/plain'] = paste(capture.output(print(obj)), collapse="\n")
-        send_response("execute_result", request, 'iopub',
-                  list(data=data, metadata=namedlist(),
-                  execution_count=execution_count))
+    handle_value <- function (obj) {
+        data <- list()
+        if (getOption('jupyter.rich_display')) {
+            for (mime in getOption('jupyter.result_mimetypes')) {
+                r <- mime2repr[[mime]](obj)
+                if (!is.null(r)) data[[mime]] <- r
+            }
+        } else {
+            data[['text/plain']] <- repr_text(obj)
```

Comment:
it’s more clear.

[[]] means: get/set one element

[] means: get/set a slice

this is important when e.g. slicing something with names:

```r
> l <- list(a = 1, b = 2)
> l[1]
$a
[1] 1
> l[[1]]
[1] 1
```

the first one will return a list that is a slice of `l`, the second one will return an element.

also consider this:

```r
> # v is a vector
> # i should be a single integer but due to a bug is c(1,2)
> x <- v[[i]]
Fehler in l[[i]] : Indizierung außerhalb der Grenzen
> x <- v[i]
> # x is expected to be one value here, but is a vector of length 2 instead
```

the first one shows the error, the second one silently propagates it.